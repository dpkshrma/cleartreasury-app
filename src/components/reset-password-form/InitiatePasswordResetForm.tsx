import { Alert, Button, Input } from "@clear-treasury/design-system";
import React, { FunctionComponent, Ref } from "react";
import styles from "./reset-password-form.module.scss";

export interface InitiatePasswordResetFormProps {
  onSubmit: () => void;
  loading: boolean;
  emailRef: Ref<HTMLInputElement | null>;
}

const InitiatePasswordResetForm: FunctionComponent<InitiatePasswordResetFormProps> =
  ({ onSubmit, emailRef, loading }) => {
    return (
      <React.Fragment>
        <div className={`mb-6 ${styles.alertContainer}`}>
          <Alert
            status={Alert.Status.PRIMARY}
            text="Please enter the email address you used to sign up and we will send you a link to reset your password"
          />
        </div>
        <div className="mb-6">
          <Input
            name="email"
            type="email"
            label="Email address"
            placeholder="Enter your email"
            ref={emailRef}
          />
        </div>
        <Button
          loading={loading}
          size={Button.Size.LARGE}
          onClick={onSubmit}
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          disabled={loading}
        >
          Send Code
        </Button>
      </React.Fragment>
    );
  };

export default InitiatePasswordResetForm;
