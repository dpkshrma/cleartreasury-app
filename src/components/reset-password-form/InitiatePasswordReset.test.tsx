import * as React from "react";
import * as nextRouter from "next/router";
import { render } from "../../../test/testUtils";
import * as Stories from "./InitiatePasswordResetForm.stories";

jest.mock("next/router");
(nextRouter.useRouter as jest.Mock).mockResolvedValue({
  isReady: true,
});

describe("InitiatePasswordResetForm", () => {
  it("matches snapshot for default state", () => {
    const { asFragment } = render(
      <Stories.Default {...Stories.Default.args} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
