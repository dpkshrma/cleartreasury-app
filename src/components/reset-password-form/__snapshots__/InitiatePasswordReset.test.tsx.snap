// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitiatePasswordResetForm matches snapshot for default state 1`] = `
<DocumentFragment>
  <form>
    <div
      class="mb-6"
    >
      <div
        class="rounded-md p-4 border border-theme-color-primary text-theme-color-primary bg-theme-color-primary-background"
      >
        <div
          class="flex items-center"
        >
          <div
            class="text-theme-color-primary h-6 w-6 flex-shrink-0"
          >
            <svg
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <p
            class="text-sm ml-3"
          >
            Please enter the email address you used to sign up and we will send you a link to reset your password
          </p>
        </div>
      </div>
    </div>
    <div
      class="mb-6"
    >
      <div
        class="w-full relative "
      >
        <label
          class="text-base text-gray-700 inline-block mb-2"
          for="email"
          id="email-label"
        >
          Email address
        </label>
        <input
          class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
          id="email"
          name="email"
          placeholder="Enter your email"
          type="email"
          value=""
        />
      </div>
    </div>
    <button
      class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
    >
      Send Code
    </button>
  </form>
</DocumentFragment>
`;
